gkr_iop/src/precompiles/lookup_keccakf.rs

```diff
@@ -670,99 +690,119 @@ impl<E: ExtensionField> ProtocolBuilder<E> for KeccakLayout<E> {
             );
         }
 
-        // TODO add rotation constrain
-
-        let mut global_and_lookup = 0;
-        let mut global_xor_lookup = 3 * AND_LOOKUPS;
-        let mut global_range_lookup = 3 * AND_LOOKUPS + 3 * XOR_LOOKUPS;
-
-        let ConstraintSystem {
-            mut expressions,
-            mut expr_names,
-            mut evals,
-            and_lookups,
-            xor_lookups,
-            range_lookups,
-            ..
-        } = system;
-
-        for (i, lookup) in chain!(and_lookups, xor_lookups, range_lookups)
-            .flatten()
-            .enumerate()
-        {
-            expressions.push(lookup);
-            let idx = if i < 3 * AND_LOOKUPS {
-                &mut global_and_lookup
-            } else if i < 3 * AND_LOOKUPS + 3 * XOR_LOOKUPS {
-                &mut global_xor_lookup
-            } else {
-                &mut global_range_lookup
-            };
-            expr_names.push(format!("round 0th: {i}th lookup felt"));
-            evals.push(lookup_outputs[*idx].clone());
-            *idx += 1;
-        }
-
-        assert_eq!(global_and_lookup, 3 * AND_LOOKUPS);
-        assert_eq!(global_xor_lookup, 3 * AND_LOOKUPS + 3 * XOR_LOOKUPS);
-        assert_eq!(global_range_lookup, LOOKUP_FELTS_PER_ROUND);
-
         let keccak_input8: ArrayView<(WitIn, EvalExpression<E>), Ix3> =
             ArrayView::from_shape((5, 5, 8), keccak_input8).unwrap();
         let keccak_input32 = keccak_input32.to_vec();
         let mut keccak_input32_iter = keccak_input32.iter().cloned();
 
-        // process keccak input
+        let keccak_output32 = keccak_output32.to_vec();
+        let keccak_output8: ArrayView<(WitIn, EvalExpression<E>), Ix3> =
+            ArrayView::from_shape((5, 5, 8), keccak_output8).unwrap();
+        let mut keccak_output32_iter = keccak_output32.iter().cloned();
+
+        // process keccak output
```

Comment:
Here I swap the order: keccak_output/keccak_input/lookup, to match with order of `allocate_output_evals` above